dask/array/core.py

```diff
@@ -1219,19 +1245,20 @@ def __array__(self, dtype=None, **kwargs):
         return x
 
     def __array_function__(self, func, types, args, kwargs):
-        import dask.array as module
+        if func not in _HANDLED_FUNCTIONS:
+            warnings.warn(
+                "`{}` is not implemented by dask, explicitly "
+                "coerce your array (e.g. with numpy.asarray) "
+                "to silence this warning".format(func.__name__)
```

Comment:
It's worth contemplating what we want the final state here to be. If we want dask to eventually give an error for these cases, then we should make this a `FutureWarning`.

Reply:
My preference is no warning, no error. I'd much prefer people to not have to change their code, plus I find `np.median(dask_array)` more user-friendly than `np.median(np.asarray(dask_array))`.

If the explicit `asarray` conversion is preferred, then I agree that `FutureWarning` is more appropriate.

Reply:
My experience as a library author who's written a tool on top of dask (namely, xarray) is that accidental coercion of dask arrays to NumPy is a serious bug-magnet. Inside xarray, we try to ensure that we never coerce dask array. If `np.median(dask_array)` worked, then it would be really hard to know when we used an unsupported operation.

Even for new users, it's not a great experience for people to start using dask and find that it's performance is awful because they are using it wrong (and nothing warned them otherwise).

I could see this being a user-configurable option, but  a warning (or eventually, an error) should probably be the default.

Reply:
Okay changed to `FutureWarning` and added "this may break in a future release"